sys/dev/nmdm/nmdm.c

```diff
@@ -115,16 +115,16 @@ nmdm_close(struct tty *tp)
 		return;
 
 	/* Shut down self. */
+	tty_lock(tp);
 	tty_rel_gone(tp);
 
 	/* Shut down second part. */
-	tty_lock(tp);
```

Comment:
This is necessary because they share a single mutex, and it needs to be held for the below `tty_rel_gone`.  I'd be in favor of switching this one to `otp` to make it more clear, though.

Reply:
"Fixed" within the realm of the second and third patch.